mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -446,6 +459,11 @@ private[ml] object DefaultParamsReader {
     val cls = Utils.classForName(metadata.className)
     cls.getMethod("read").invoke(null).asInstanceOf[MLReader[T]].load(path)
   }
+
+  def loadInitialModel[M <: Model[M]](path: String, sc: SparkContext): Try[M] = {
+    val initialModelPath = new Path(path, "initialModel").toString
+    Try(loadParamsInstance[M](initialModelPath, sc))
```

Comment:
I don't think the overhead is that large, and if there is no model then it should fail fast as the directory won't exist.